matlab/src/Util.h

```diff
@@ -60,4 +60,32 @@ void* createShared(std::shared_ptr<T> p)
     return new std::shared_ptr<T>(std::move(p));
 }
 
+inline void* createProxy(Ice::ObjectPrx p)
+{
+    return new Ice::ObjectPrx(std::move(p));
+}
+
+inline void* createProxy(std::optional<Ice::ObjectPrx> p)
+{
```

Comment:
Would be clear to pass as `const&` we never store p, or pass it to other methods

Reply:
The implementation moves p, see below.

The general rule is:
 - if the implementation "keeps" the arg, pass it by value
 - if the implementation only reads the arg, pass it by const&

Reply:
In particular, `.value()` below corresponds to the && overload:
https://en.cppreference.com/w/cpp/utility/optional/value (2)

Reply:
The implementation moves `p`, and calls `.value()` on the returned `rrefence`.